src/server/stat_job.c

```diff
@@ -378,12 +378,16 @@ status_subjob(job *pjob, struct batch_request *preq, svrattrl *pal, int subj, pb
 	if ((pjob->ji_qs.ji_svrflags & JOB_SVFLG_ArrayJob) == 0)
 		return PBSE_IVALREQ;
 
-	/* if subjob is running, use real job structure */
-
-	if (get_subjob_state(pjob, subj) == JOB_STATE_RUNNING) {
-		psubjob = find_job(mk_subjob_id(pjob, subj));
-		if (psubjob)
-			status_job(psubjob, preq, pal, pstathd, bad);
+	/* if subjob job obj exists, use real job structure */
+
+	psubjob = find_job(mk_subjob_id(pjob, subj));
+	if (psubjob) {
+		if((pjob->ji_qs.ji_state == JOB_STATE_FINISHED) && strchr(preq->rq_extend, (int)'x')) {
+			psubjob->ji_qs.ji_state = JOB_STATE_FINISHED;
```

Comment:
set the state of subjob using the setter function like set_attr_svr().

Reply:
fixed in rc 10 https://github.com/PBSPro/pbspro/pull/590/commits/2d01ac8644233b82ef5ee900e7a56132e5c95998